src/mds/DamageTable.cc

```diff
@@ -200,6 +201,29 @@ bool DamageTable::notify_remote_damaged(inodeno_t ino, std::string_view path)
   return false;
 }
 
+void DamageTable::remove_dentry_damage_entry(CDir *dir)
+{
+  const std::map<DentryIdent, DamageEntryRef> &frag_dentries =
+    dentries.at(DirFragIdent(dir->inode->ino(), dir->frag));
+    for (const auto &i : by_id){
+      for (const auto &dn : frag_dentries){
+        if (i.second == dn.second){
+          DamageTable::erase(i.first);
+        }
+      }
+    }
+}
+
+void DamageTable::remove_dirfrag_damage_entry(CDir *diri)
+{
+  DamageTable::erase(dirfrags.find(DirFragIdent(diri->inode->ino(), diri->frag))->second->id);
```

Comment:
what if the directory inode hasn't been added to the `dirfrags` in DamageTable ?

also, the argument name should just be `dir` and not `diri` ... since `diri` implies the directory inode and not a directory

Reply:
> what if the directory inode hasn't been added to the `dirfrags` in DamageTable ?
There is check for the `damage_id` in the `DamageTable::erase()` but to ensure and reduce the iteration prior to that in case of its absence, i will put a check before that.